src/main/java/com/iridium/iridiumteams/database/IridiumUser.java

```diff
@@ -54,11 +54,19 @@ public Player getPlayer() {
         return Bukkit.getServer().getPlayer(uuid);
     }
 
-    public boolean canFly(IridiumTeams<T, ?> iridiumTeams) {
+    public boolean canFly(Optional<T> team, IridiumTeams<T, ?> iridiumTeams) {
```

Comment:
Why are we passing Optional<T> team in here?

Reply:
because if the team doesnt exist, it checks the user's team, and if *that* doesn't exist, then it doesn't pass the team permission check

Reply:
But we arnt using it anywhere in this method?

Reply:
we dont need to do that for a switch? its just asking what the player gamemode is and running down the list from the point it finds the value that relates to it - meaning we find the value *once* and then just do something based on that

Reply:
i can still change it if you really want me to but its less words 🤷

Reply:
wait im fucking stupid, wrong suggestion lmao

Reply:
we use it in ``canFly``, since i changed the method signature to use it